charts/tf-controller/templates/runner-discovery-service.yaml

```diff
@@ -0,0 +1,16 @@
+{{- range include "tf-controller.runner.allowedNamespaces" . | fromJsonArray }}
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: tf-runner
+  namespace: {{ . }}
+spec:
+  clusterIP: None
+  ports:
+  - name: grpc
+    port: 30000
+  selector:
+    app.kubernetes.io/created-by: tf-controller
+		app.kubernetes.io/name: tf-runner
```

Comment:
```suggestion
  selector:
    app.kubernetes.io/created-by: tf-controller
    app.kubernetes.io/name: tf-runner
```

Reply:
Please introduce the new chart value called `usePodSubdomainResolution` to
1. add `--use-pod-subdomain-resolution` flag to the controller (which requires you to create that flag for it too)
2. guard generation of `Service` objects here.

Please note that we need `tf-runner` Service by default for `flux-system` namespace too.

Reply:
Modifications performed :smile: